lib/serverless/api-gateway.js

```diff
@@ -107,56 +110,82 @@ function normalizeHeaders(event, lowerCaseKey = false) {
  *                    to create a web transaction.
  */
 function isLambdaProxyEvent(event) {
-  return isGatewayV1Event(event) || isGatewayV2Event(event)
+  return isGatewayV1Event(event) || isGatewayV2Event(event) || isAlbEvent(event)
 }
 
+/**
+ * Iterates over the minimum signature properties of an event triggered by API Gateway
+ * to determine the version. V1 and V2 have a lot of overlap, but some signature differences.
+ *
+ * @param {object} targetEvent The event to inspect.
+ * @param {Array} searchFor An array of keys unique to this proxy type
+ * @returns {boolean} Whether this event has matches for the keys we're checking
+ */
+function findProperties(targetEvent, searchFor) {
+  const keys = Object.keys(targetEvent)
+  for (const el of searchFor) {
+    if (keys.indexOf(el) > -1) {
+      return true
+    }
+  }
+  return false
+}
+
+/**
+ * We need to determine whether this invocation event was triggered from a proxy: API Gateway V1, V2, ALB, or some other service.
+ * API Gateway V1/V2 and ALB events aren't guaranteed always to have the same properties in all cases. There are, however, properties
+ * that are vary consistently depending on the service triggering the event.
+ *
+ * API Gateway v2 HTTP
+ * API Gateway v1 HTTP: top-level httpMethod, resource, multiValueHeaders (possibly), multiValueQueryStringParameters (possibly)
+ * API Gateway v1 REST same as HTTP, but without version
+ * ALB: very similar to API Gateway v1, but requestContext contains an elb property
+ */
```

Comment:
What is this docblock style comment meant to be attached to?

Reply:
This content best fits with the function that tests for these properties, so I'll consolidate that content.